src/lib/reasoners/ite_rel.ml

```diff
@@ -81,6 +97,10 @@ let add_to_guarded p s t mp =
   let st = try ME.find p mp with Not_found -> SE2.empty in
   ME.add p (SE2.add (s, t) st) mp
 
+(* Check if the condition of the ite t is an assumed predicate. If do so,
```

Comment:
What does "an assumed predicate" here? Is it "a predicate entailed by the current state"?

Reply:
I am not totally sure. We should check what is the semantic of the assume function in `Sig_rel.RELATION`. By assumed predicate, I mean predicate that appears once in the argument of the assume function in `Ite_rel`.

Reply:
OK, yeah, I *think* that is the case then but not 100% sure. Maybe add the info that "assumed predicate" means "predicate that appears in the argument of `assume`", that is already useful, and we will understand when `assume` is called later then.

Reply:
More precisely I believe that the predicates that `assume` gets called on are predicates entailed by the current state, but I do not know whether all predicates that have been either asserted or learnt result in a call to `assume`.